### DISCUSSION TITLE
部署在docker上后，在路由列表上显示“路由器不可用”

### DISCUSSION THREAD

**Discussion by KazeLiu** [Mirouterui/mirouter-ui#43] [Q&A]
家里的拓扑是这样的：光猫->主路由（192.168.2.1）-mesh->子路由（192.168.2.175）->软路由(192.168.2.199)

路由器的型号是 Xiaomi路由器BE3600 2.5G版 
系统版本 MiWiFi 稳定版 1.0.68
前端版本 v1.5.0-2
后端版本 

现在在软路由上的docker部署这个项目，然后这是我的配置文件：
```
{
  "dev": [
    {
      "password": "",
      "key": "a2ffa5c9be07488bbb04a3a47d3c5f6a",
      "ip": "192.168.2.1",
      "routerunit": true
    }
  ],
  "history": {
    "enable": true,
    "maxsaved": 60,
    "sampletime": 86400
  },
  "debug": true,
  "port": 6789,
  "tiny": false,
  "flushTokenTime": 1800,
  "netdata_routernum": 0
}
```
打开项目后，在路由列表内的结果是路由器不可用，首页全部都连接不上。
![image](https://github.com/user-attachments/assets/c29598ed-04f8-42c6-813d-8b21a466a08c)
![image](https://github.com/user-attachments/assets/d2cfa66d-5718-4936-9f45-b8d2bbb565de)

我还有个疑惑，就是我一直不清楚文档里面说的“后端”是什么东西，我在界面上也发现了说需要填写后端ip和端口，我在项目内也发现了后端的安装包，但是我不知道怎么安装这个“后端”，我也不知道是不是因为没有安装这个后端导致我无法获取数据。

**Comment by thun888** [comment]
一个一个来：

1. 后端和前端：后端即程序本体，前端就是你在浏览器中看到的网页。只是说程序还能同时提供网页服务，让你访问到前端。这个配置项是为了使用在线前端时才配置，以使用启用tiny模式后不提供该网页服务的后端。既然你已经运行了，那就说明已经部署了后端了。确实可能会引发误解，下个版本会改。
2. "routerunit"参数：意为是否在小米路由器上运行，你在软路由上运行自然不需要设置为true了。这个组合词也确实起得不好，我想想以后怎么改
![image](https://github.com/user-attachments/assets/dcd3b128-77fd-4a88-9903-62185b5ebfdf)

3. 路由器不可用：程序在获取token时会先检测路由器是否可用，若五秒内路由器无响应则标记为不可用。你可以尝试执行该命令检测在容器里能否访问到路由器

> docker exec mirouter-ui ping 192.168.2.1

![image](https://github.com/user-attachments/assets/2fb122a4-113a-48a6-b64b-2bc79688b48d)
 
最后，我还是建议自己检查网络环境，这应该不是本程序的问题
